src/Sylius/Bundle/ShopBundle/Tests/Calculator/OrderItemsSubtotalCalculatorTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ShopBundle\Tests\Calculator;
+
+use Mockery;
+use PHPUnit\Framework\TestCase;
+use Sylius\Bundle\ShopBundle\Calculator\OrderItemsSubtotalCalculator;
+use Sylius\Bundle\ShopBundle\Calculator\OrderItemsSubtotalCalculatorInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+
+final class OrderItemsSubtotalCalculatorTest extends TestCase
```

Comment:
Perhaps another time. I'm interested to know when do you prefer spec over unit when it comes to testing? Very similar results can be achieved between the two, but I guess the focus does not 100% have to be on structure alone for unit tests?

Reply:
PHPSpec - unit testing, but to be more precise it is more for designing implementation. Testing every class in total separation, all collaborators are mocked and you need to be precise, what should be asserted (shouldBeCalled) and return (willReturn). Prophecy is a great part of this library, which was recently deprecated on PHPUnit. Also, it forces you for the usage of mocks rather than separated classes.
PHPUnit - functional testing - mostly used for more than one class initialized in the test suite. Some examples, testing of services fetched from the container, FormType testing, testing of configuration and old API testing
Behat - acceptance tests, focused mostly on business value and covered through all of Sylius layers

The distinction between PHPSpec & PHPUnit in terms of unit testing can indeed be a matter of taste, but we've decided to go with PHPSpec, and keeping codebase coherent is important for us.

Is it clearer right now?

Reply:
Thanks, that's pretty clear. I'd hate to jump in and learn about how phpspec works right here and now, but next time I'll do it that way for sure.